src/lib/io/include/sol-uart.h

```diff
@@ -224,21 +239,35 @@ struct sol_uart *sol_uart_open(const char *port_name, const struct sol_uart_conf
 void sol_uart_close(struct sol_uart *uart);
 
 /**
- * @brief Perform a UART asynchronous transmission.
+ * @brief Perform an UART asynchronous transmission.
+ *
+ * @param uart The UART bus handle
+ * @param blob The blob to be written
+ * @return 0 on success negative errno on error
+ *
```

Comment:
beef up this doc. Say that this function will take a reference of the blob and right before the blob is automatically unreferrenced (remove the extra reference), `tx_cb` is called so users can know when a specific blob was fully written. Say that you can query pending amount of bytes with `sol_uart_get_pending_bytes()` in cases where the producer is faster than the UART.

Also link this to a general doxygen section on this pattern. Then link this with crypto `sol_message_crypto` and http progressive (when it is converted). This will help us to ensure the pattern is reused in the future (and keep an eye when people send PR with patterns matching this one, you'll be responsible to keep consistency!)